### PR TITLE
ci: skip mgr pod restart count upgrade 1.22.x suite

### PR DESCRIPTION
**Pull request by subhamkrai**
for now, let's skip the mgr pod restart count
for upgrade suite 1.22.x to get the CI green
and so that we don't skip any other error in name
of mgr restart count.

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by travisn** [comment]
Looks like the `multus-cluster-network` canary test started failing around the same time as the upgrade suite. None of the merges around that time look related to the failures for that multus test or the upgrade suite failures, so I wonder if it's some environmental issue change in the runners that is causing the failures.

**Comment by travisn** [comment]
FWIW, when connecting to tmate in this [upgrade test CI run](https://github.com/rook/rook/actions/runs/5720206213/job/15499679483?pr=12631), I saw the following in the operator log. It seems the operator is stuck on a cancelled context. This was while the test was waiting for the first mon to be upgraded from Ceph v16 to v17, so it seems related to the update of the cluster CR. 

```
2023-07-31 22:58:15.978787 E | util: error occurred waiting for deployments to be updated. retrying after 2s. failed to list deployments: client rate limiter Wait returned an error: context canceled
2023-07-31 22:58:15.978815 D | util: waiting for deployments to be updated. retrying after 2s
2023-07-31 22:58:16.412047 E | util: error occurred waiting for deployments to be updated. retrying after 2s. failed to list deployments: client rate limiter Wait returned an error: context canceled
2023-07-31 22:58:16.412076 D | util: waiting for deployments to be updated. retrying after 2s
2023-07-31 22:58:17.978976 E | util: error occurred waiting for deployments to be updated. retrying after 2s. failed to list deployments: client rate limiter Wait returned an error: context canceled
2023-07-31 22:58:17.978999 D | util: waiting for deployments to be updated. retrying after 2s
2023-07-31 22:58:18.412236 E | util: error occurred waiting for deployments to be updated. retrying after 2s. failed to list deployments: client rate limiter Wait returned an error: context canceled
2023-07-31 22:58:18.412317 D | util: waiting for deployments to be updated. retrying after 2s
2023-07-31 22:58:19.979810 E | util: error occurred waiting for deployments to be updated. retrying after 2s. failed to list deployments: client rate limiter Wait returned an error: context canceled
2023-07-31 22:58:19.979840 D | util: waiting for deployments to be updated. retrying after 2s
2023-07-31 22:58:20.416075 E | util: error occurred waiting for deployments to be updated. retrying after 2s. failed to list deployments: client rate limiter Wait returned an error: context canceled
2023-07-31 22:58:20.416097 D | util: waiting for deployments to be updated. retrying after 2s
```

But other upgrade tests don't have this error consistently, so either we have multiple issues, or this is not the root cause.

**Comment by subhamkrai** [comment]
> FWIW, when connecting to tmate in this [upgrade test CI run](https://github.com/rook/rook/actions/runs/5720206213/job/15499679483?pr=12631), I saw the following in the operator log. It seems the operator is stuck on a cancelled context. This was while the test was waiting for the first mon to be upgraded from Ceph v16 to v17, so it seems related to the update of the cluster CR.
> 
> ```
> 2023-07-31 22:58:15.978787 E | util: error occurred waiting for deployments to be updated. retrying after 2s. failed to list deployments: client rate limiter Wait returned an error: context canceled
> 2023-07-31 22:58:15.978815 D | util: waiting for deployments to be updated. retrying after 2s
> 2023-07-31 22:58:16.412047 E | util: error occurred waiting for deployments to be updated. retrying after 2s. failed to list deployments: client rate limiter Wait returned an error: context canceled
> 2023-07-31 22:58:16.412076 D | util: waiting for deployments to be updated. retrying after 2s
> 2023-07-31 22:58:17.978976 E | util: error occurred waiting for deployments to be updated. retrying after 2s. failed to list deployments: client rate limiter Wait returned an error: context canceled
> 2023-07-31 22:58:17.978999 D | util: waiting for deployments to be updated. retrying after 2s
> 2023-07-31 22:58:18.412236 E | util: error occurred waiting for deployments to be updated. retrying after 2s. failed to list deployments: client rate limiter Wait returned an error: context canceled
> 2023-07-31 22:58:18.412317 D | util: waiting for deployments to be updated. retrying after 2s
> 2023-07-31 22:58:19.979810 E | util: error occurred waiting for deployments to be updated. retrying after 2s. failed to list deployments: client rate limiter Wait returned an error: context canceled
> 2023-07-31 22:58:19.979840 D | util: waiting for deployments to be updated. retrying after 2s
> 2023-07-31 22:58:20.416075 E | util: error occurred waiting for deployments to be updated. retrying after 2s. failed to list deployments: client rate limiter Wait returned an error: context canceled
> 2023-07-31 22:58:20.416097 D | util: waiting for deployments to be updated. retrying after 2s
> ```
> 
> But other upgrade tests don't have this error consistently, so either we have multiple issues, or this is not the root cause.

I'm not sure if this is the root cause, but the reason test is failing is due to mgr pod restart and also it is failing to write to pod in some validation for filesystem. So, first I'm trying to fix the issue to write date into pod

**Review by travisn — CHANGES_REQUESTED** [review]
The 1.22 test actually failed again with the error of a pod restarting, but I don't see which pod. Let's figure that out before merging.

**Comment by subhamkrai** [comment]
> It's concerning to skip this failure, but getting green CI is necessary first, and we will track the issue with #12646.

I agree but I'm afraid we may skip another issue by thinking that the upgrade suite is failing already. So, I want to get it green

**Review by travisn — APPROVED** [review]

File: .github/workflows/integration-test-upgrade-suite.yaml [hunk 1]
```diff
@@ -63,7 +63,7 @@ jobs:
         run: |
           tests/scripts/github-action-helper.sh collect_udev_logs_in_background
           export DEVICE_FILTER=$(lsblk|awk '/14G/ {print $1}'| head -1)
-          go test -v -timeout 1800s -failfast -run CephUpgradeSuite/TestUpgradeRook github.com/rook/rook/tests/integration
+          go test -v -timeout 2400s -failfast -run CephUpgradeSuite/TestUpgradeRook github.com/rook/rook/tests/integration
```
Comment (travisn) [inline .github/workflows/integration-test-upgrade-suite.yaml:66]: We also need this change in integration-tests-on-release.yaml for the master/release CI.

Comment (subhamkrai) [reply .github/workflows/integration-test-upgrade-suite.yaml:66]: fixed #12656

**Pull request merged**